CTP/shell/src/com/navercorp/cubridqa/shell/deploy/Deploy.java

```diff
@@ -60,16 +60,23 @@ public Deploy(Context context, String currEnvId, boolean laterJoined) throws Exc
 	
 	public void deploy() throws Exception {
 		context.getFeedback().onDeployStart(envIdentify);
+		ArrayList<String> relatedHosts = context.getRelatedHosts(currEnvId);
```

Comment:
1. Move configureForHAMode into new Class: DeployHA.java
2. It's better to deploy HA instance for the first related host
3. Remove HAMode property in Context.java in order to keep more general for it.

To be:
```java
ArrayList<String> relatedHosts = context.getRelatedHosts(currEnvId);
if(relatedHosts !=null && relatedHosts.size() >0) {
        DeployHA  d = new DeployHA (context, masterEnvId, relatedHosts.get(0), log).deploy();
        d.close();
}
```